File: rpgext/reforge/src/main/java/cc/mewcraft/reforge/command/PluginCommands.java
package cc.mewcraft.reforge.command;

import cc.mewcraft.reforge.ReforgePlugin;
import com.google.inject.Inject;
import com.google.inject.Injector;
import com.google.inject.Singleton;

@Singleton
public class PluginCommands {
    private final Injector injector;
    private final ReforgePlugin plugin;
    private final CommandRegistry registry;

    @Inject
    public PluginCommands(
        Injector injector,
        ReforgePlugin plugin,
        CommandRegistry registry
    ) {
        this.injector = injector;
        this.plugin = plugin;
        this.registry = registry;
    }

    public void registerCommands() {
        // Prepare commands
        registry.prepareCommand(registry
            .commandBuilder("reforge")
            .literal("reload")
            .permission("reforge.command.reload")
            .handler(ctx -> {
                plugin.reloadConfig();
            }).build());

        // Register commands
        registry.registerCommands();
    }
}

File: worldreset/bukkit/master/src/main/kotlin/cc/mewcraft/worldreset/command/PluginCommands.kt
package cc.mewcraft.worldreset.command

import cc.mewcraft.mewcore.command.SimpleCommands
import cc.mewcraft.worldreset.manager.ServerLock
import cloud.commandframework.arguments.standard.BooleanArgument
import org.bukkit.plugin.Plugin

class PluginCommands(
    plugin: Plugin,
    private val serverLocks: ServerLock,
) : SimpleCommands<Plugin>(plugin) {
    override fun prepareAndRegister() {
        registry.prepareCommand(
            registry.commandBuilder("worldreset")
                .literal("serverlock")
                .argument(BooleanArgument.builder("status"))
                .permission("worldreset.command.admin")
                .handler { ctx ->
                    val status = ctx.get<Boolean>("status")
                    serverLocks.setLock(status)
                    ctx.sender.sendRichMessage("Current server lock: ${status.toString().uppercase()}")
                }.build()
        )
        registry.prepareCommand(
            registry.commandBuilder("worldreset")
                .literal("reload")
                .permission("worldreset.command.admin")
                .handler { ctx ->
                    plugin.onDisable()
                    plugin.onEnable()
                    ctx.sender.sendRichMessage("WorldReset has been reloaded!")
                }.build()
        )

        registry.registerCommands()
    }
}